src/draw/nxp/g2d/lv_draw_g2d_fill.c

```diff
@@ -0,0 +1,185 @@
+                               struct g2d_buf * src_buf, struct g2d_surface * src_surf);
+
+static void _g2d_set_src_surf(struct g2d_surface * src_surf, struct g2d_buf * buf, const lv_area_t * area,
+                              int32_t stride, lv_color_t color, lv_opa_t opa);
+
+static void _g2d_set_dst_surf(struct g2d_surface * dst_surf, struct g2d_buf * buf, const lv_area_t * area,
+                              int32_t stride, lv_color_t color);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_draw_g2d_fill(lv_draw_g2d_unit_t * draw_g2d_unit, const lv_draw_fill_dsc_t * dsc, const lv_area_t * coords)
+{
+    if(dsc->opa <= (lv_opa_t)LV_OPA_MIN)
+        return;
+
+    lv_layer_t * layer = draw_g2d_unit->base_unit.target_layer;
+    lv_draw_buf_t * draw_buf = layer->draw_buf;
+
+    lv_area_t rel_coords;
+    lv_area_copy(&rel_coords, coords);
+    lv_area_move(&rel_coords, -layer->buf_area.x1, -layer->buf_area.y1);
+
+    lv_area_t rel_clip_area;
+    lv_area_copy(&rel_clip_area, draw_g2d_unit->base_unit.clip_area);
+    lv_area_move(&rel_clip_area, -layer->buf_area.x1, -layer->buf_area.y1);
+
+    lv_area_t blend_area;
+    if(!lv_area_intersect(&blend_area, &rel_coords, &rel_clip_area))
+        return; /*Fully clipped, nothing to do*/
+
+    /* G2D takes stride in pixels. */
+    int32_t stride = draw_buf->header.stride / (lv_color_format_get_bpp(draw_buf->header.cf) / 8);
+
+    /* Destination buffer */
+    struct g2d_buf * dst_buf = lv_search_buf_map(draw_buf->data);
+
+    bool has_opa = (dsc->opa < (lv_opa_t)LV_OPA_MAX);
+    struct g2d_surface * dst_surf = lv_malloc(sizeof(struct g2d_surface));
+    G2D_ASSERT_MSG(dst_surf, "Failed to alloc destination surface.");
+    _g2d_set_dst_surf(dst_surf, dst_buf, &blend_area, stride, dsc->color);
+
+    if(has_opa) {
+        struct g2d_buf * tmp_buf = g2d_alloc(lv_area_get_width(&blend_area) * lv_area_get_height(&blend_area) * sizeof(
+                                                 lv_color32_t), 1);
+        G2D_ASSERT_MSG(tmp_buf, "Failed to alloc temporary buffer.");
+        struct g2d_surface * src_surf = lv_malloc(sizeof(struct g2d_surface));
+        G2D_ASSERT_MSG(src_surf, "Failed to alloc source surface.");
+        _g2d_set_src_surf(src_surf, tmp_buf, &blend_area, stride, dsc->color, dsc->opa);
+        _g2d_fill_with_opa(draw_g2d_unit->g2d_handle, dst_buf, dst_surf, tmp_buf, src_surf);
+        g2d_free(tmp_buf);
+        lv_free(src_surf);
+    }
+    else {
+        _g2d_fill(draw_g2d_unit->g2d_handle, dst_buf, dst_surf);
+    }
+
+    lv_free(dst_surf);
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+void _g2d_set_src_surf(struct g2d_surface * src_surf, struct g2d_buf * buf, const lv_area_t * area, int32_t stride,
```

Comment:
```suggestion
/**********************
 *   STATIC FUNCTIONS
 **********************/

static void _g2d_set_src_surf(struct g2d_surface * src_surf, struct g2d_buf * buf, const lv_area_t * area, int32_t stride,
```

And all the others, please